sklearn/ensemble/weight_boosting.py

```diff
@@ -1045,9 +1034,10 @@ def _boost(self, iboost, X, y, sample_weight, random_state):
         y_predict = estimator.predict(X)
 
         error_vect = np.abs(y_predict - y)
-        error_max = error_vect.max()
+        sample_mask = sample_weight > 0
+        error_max = error_vect[sample_mask].max()
```

Comment:
maybe do
```
masked_sample_weight = sample_weight[sample_mask]
masked_error_vect = error_vect[sample_mask]
```
and the use those instead of doing the masking several times.